gen/calc.py

```diff
@@ -224,6 +224,17 @@ def validate_host_list(host_list):
         return host_list
     assert not any(azure_format_check), "Azure static master list and IP based static master list cannot be mixed. Use "
     "either all Azure IP references or IPv4 addresses."
+
```

Comment:
this should be a separate function. Technically duplicate IPs with different ports are fine. I'm not a fan of making a function that radically changes behavior based on a parameter.